main.go

```diff
@@ -10,8 +11,12 @@ func main() {
 	//make an AmqpPublisher to be injected into the following methods
 	amqpPublisher := new(utils.AmqpPublisher)
 
+	//close mongo session
+	defer db.MgoSession.Close()
```

Comment:
Where does this db.MgoSession get opened? Its init function shows no usages. Is it implicit?

Reply:
This closes the main mongo session in the db/db.go file.

Reply:
dangling closes like this are dangerous, you should close in the same file the connection was opened in.

Reply:
connection is opened and closed in the same file